Sources/Utils/Backtrace.cpp

```diff
@@ -67,13 +69,76 @@ void PrintBacktrace() {
   }
 }
 #elif defined(OS_WIN32)
+
+static HANDLE CurrentProcess() {
+  // Based on
+  // https://learn.microsoft.com/en-us/windows/win32/debug/initializing-the-symbol-handler
+  HANDLE process = GetCurrentProcess();
+  // Explanation for this is at
+  // https://github.com/apple/swift/blob/0c56f1468c0f56ab9b96464ba0d63f7db4b84b69/stdlib/public/runtime/ImageInspectionCOFF.cpp#L42-L53.
+  if (!DuplicateHandle(process, process, process, &process, 0, false,
+                       DUPLICATE_SAME_ACCESS)) {
+    DS2LOG(Error, "failed to dup current process handle; proceeding anyway");
+  }
+  if (SymInitialize(process, NULL, TRUE)) {
+    return process;
+  }
+  // SymInitialize failed
+  auto error = GetLastError();
+  DS2LOG(Error, "SymInitialize returned error : %lu", error);
+  return HANDLE{}; // Unreachable.
+}
+
+static std::string SymbolForAddress(HANDLE currentProcess, void *pc) {
+  DWORD64 address = reinterpret_cast<DWORD64>(pc);
+  // Based on
+  // https://learn.microsoft.com/en-us/windows/win32/debug/retrieving-symbol-information-by-address
+  char buffer[sizeof(SYMBOL_INFO) + MAX_SYM_NAME * sizeof(TCHAR)];
+  PSYMBOL_INFO symbol = (PSYMBOL_INFO)buffer;
+  symbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+  symbol->MaxNameLen = MAX_SYM_NAME;
+  DWORD64 offset = 0;
+  if (!SymFromAddr(currentProcess, address, &offset, symbol)) {
+    auto error = GetLastError();
+    DS2LOG(Error, "SymFromAddr returned error : %lu", error);
+    return "<unknown>";
+  }
+  std::ostringstream os;
+  os << symbol->Name << " +0x" << std::hex << offset << std::dec;
+
+  IMAGEHLP_LINE64 line = {};
+  DWORD lineOffset = 0;
+  line.SizeOfStruct = sizeof(line);
+  if (SymGetLineFromAddr64(currentProcess, address, &lineOffset, &line)) {
+    os << " (" << line.FileName << ":" << line.LineNumber << ")";
+  } else {
+    auto error = GetLastError();
+    if (error == ERROR_INVALID_ADDRESS) {
+      // This is not an error, it just means that the address is not in a
+      // module that has line number information.
+      DS2LOG(Debug, "SymGetLineFromAddr64 returned error: %lu", error);
+    } else {
+      DS2LOG(Error, "SymGetLineFromAddr64 returned error: %lu", error);
+      os << " (<missing>)";
+    }
```

Comment:
```suggestion
    auto error = GetLastError();
    DS2LOG(Debug, "SymGetLineFromAddr64 returned error: %lu", error);
    if (error != ERROR_INVALID_ADDRESS)
      os << " (<missing>)";
```

Reply:
My version emits Debug for expected and Error for unexpected cases. I'd like to keep that.

Reply:
Could do a ternary to select the level.